src/main/java/mate/academy/dao/impl/CinemaHallDaoImpl.java

```diff
@@ -0,0 +1,61 @@
+package mate.academy.dao.impl;
+
+import java.util.List;
+import java.util.Optional;
+import mate.academy.dao.CinemaHallDao;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.CinemaHall;
+import mate.academy.util.HibernateUtil;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+
+@Dao
+public class CinemaHallDaoImpl implements CinemaHallDao {
+    private final SessionFactory factory = HibernateUtil.getSessionFactory();
+
+    @Override
+    public CinemaHall add(CinemaHall cinemaHall) {
+        Transaction transaction = null;
+
+        try (Session session = factory.openSession()) {
+            try {
+                transaction = session.beginTransaction();
+                session.persist(cinemaHall);
+                transaction.commit();
+            } catch (Exception e) {
+                if (transaction != null) {
+                    transaction.rollback();
+                }
+
+                throw new DataProcessingException(String.format("Can't add "
+                        + "cinema hall: %s to DB", cinemaHall), e);
+            }
+        }
+
+        return cinemaHall;
+    }
+
+    @Override
+    public Optional<CinemaHall> get(Long id) {
+        try (Session session = factory.openSession()) {
+            return Optional.of(session.get(CinemaHall.class, id));
+        } catch (Exception e) {
+            throw new DataProcessingException("Can't get "
+                    + "cinema hall with id: " + id, e);
+        }
+    }
+
+    @Override
+    public List<CinemaHall> getAll() {
+        try (Session session = factory.openSession()) {
+            Query<CinemaHall> getAllCinemaHallQuery
+                    = session.createQuery("from CinemaHall", CinemaHall.class);
+            return getAllCinemaHallQuery.getResultList();
+        } catch (Exception e) {
+            throw new DataProcessingException("DB is empty", e);
```

Comment:
I doubt that exception is thrown if DB is empty. Generally, empty result list is a valid response, so change the message.